### PR TITLE
MSL-617: Combine validate-requirements.sh and basic-dep-setup.sh to g…

### PR DESCRIPTION
**Pull request by kenzananram88**
…et closer to a one button install

### CODE REVIEW

File: bin/provision/validate-requirements.sh [hunk 1]
```diff
@@ -135,8 +139,12 @@ function verifyNode {
       echo found nodejs executable in PATH
       _node=node
   else
-      echo "Please install nodejs version ${minVersion} or greater"
-      exit 1;
+      if [[ "yes" == $(askYesOrNo "Would you like us to install node for you?") ]]; then
+        bash basic-dep-setup.sh -npm
```
Comment (bschwanitz) [inline bin/provision/validate-requirements.sh]: Not different from installing npm, above. Cut-and-paste error?

File: bin/provision/basic-dep-setup.sh [hunk 2]
```diff
@@ -153,21 +199,38 @@ function install_cassandra {
...
-    echo "asciidoctor Found In \$PATH"
+
+  if [[ ${MAVEN} -eq 0 ]]; then
+    command -v mvn >/dev/null && echo "mvn Found In \$PATH" || install_maven
+  fi
+  if [[ ${NPM} -eq 0 ]]; then
+    command -v npm >/dev/null && echo "npm Found In \$PATH" || install_npm
   fi
+  if [[ ${NVM} -eq 0 ]]; then
+    command -v nvm >/dev/null && echo "nvm Found In \$PATH" || install_nvm
```
Comment (bschwanitz) [inline bin/provision/basic-dep-setup.sh]: Will this work if nvm was installed via homebrew?

File: bin/provision/validate-requirements.sh (hunk 1 above)
Comment (kenzananram88) [inline bin/provision/validate-requirements.sh]: noup, its handled in the same block in the basic-dep-setup.sh. Both node and npm are installed together. So if it was installed by the previous function it shouldn't be falling here

File: bin/provision/basic-dep-setup.sh (hunk 2 above)
Comment (bschwanitz) [inline bin/provision/basic-dep-setup.sh]: Since you didn't install nvm via homebrew, don't worry about this one. I'll test after the merge.

**Pull request merged**